src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -830,6 +849,47 @@ public void damage(double amount, Entity source, DamageCause cause) {
         setLastDamager(source);
     }
 
+    /**
+     * Checks if the incoming source of damage is a "Player Hit".
+     * @param source The incoming source of damage
+     */
+    private boolean isPlayerHit(Entity source) {
+        // If directly damaged by a player
+        if (source instanceof GlowPlayer) {
+            return true;
+        // If damaged by a TNT ignited by a player
+        } else if (source instanceof GlowTntPrimed) {
+            return
+                ((GlowTntPrimed)source).getPlayer() != null
```

Comment:
Casts should have a space after the closing parentheses. There's a couple to fix here.

Ex.: `((GlowTntPrimed) source).getPlayer() != null`

Also you can store the GlowTntPrimed instance before returning because you are casting it multiple times:

```java
GlowTntPrimed tnt = (GlowTntPrimed) source;
return tnt.getPlayer() != null [...]
```